### DISCUSSION TITLE
Virtualized lists

### DISCUSSION THREAD

**Discussion by rolanday** [adobe/react-spectrum#5056] [Q&A] [answered]
Hello,

I understand react-stately supports visualized lists, as covered [here](https://react-spectrum.adobe.com/react-stately/collections.html) -- and I assume this applies to both react-aria hooks and RAC (even though RAC docs make no mention of virtualized).

A couple questions.

1. Is virtualization enabled by default for GridList and ListBox (hooks and RAC version)?
2. Are there any important limitations to be aware of, such as all items need to be same height?

Thanks!

P.S. Congrats on RAC hitting beta

**Comment by snowystinger** [comment, marked as answer]
Hey there, and thanks!

Our hooks can support virtualization, however, they do not do virtualization for you because they do not control rendering. There are a few threads and discussions in our repo on this topic: https://github.com/adobe/react-spectrum/discussions/4367, https://github.com/adobe/react-spectrum/issues/998

Our RAC docs do not make mention of virtualizing yet because we do not think we support that yet. It's on our radar though.

**Reply by ghost** [reply]
Hey there, any update on where virtualized lists are as of today? Thank you very much! 🙏

**Reply by snowystinger** [reply]
Thanks for the ping, nothing changed yet, but we're hoping to get to it in the first half of the year.

**Reply by hwb15** [reply]
Saw @devongovett's tweet the other day which mentioned some virtualized scrolling(https://x.com/devongovett/status/1784366332143968609) and saw today the massive release went out, although don't think anything is there just yet for collections as no commits so maybe in pipeline for next release?

**Reply by devongovett** [reply]
Yeah follow #5912 for updates on that. Will be continuing work on it soon.

**Reply by hwb15** [reply]
Legend thank you will do!

**Reply by ghost** [reply]
That's great news, thanks so much!

**Comment by rolanday** [comment]
Thanks for getting back, noted, and helps to clarify. Also very much appreciate your and Adobe team's responsiveness to questions on this "channel".

**Comment by reidbarber** [comment]
Virtualizer is now documented: https://react-spectrum.adobe.com/react-aria/Virtualizer.html
